spec/controllers/admin/users_controller_spec.rb

```diff
@@ -0,0 +1,167 @@
+      expect(assigns(:user)).to eq(user)
+    end
+  end
+
+  describe "POST #create" do
+    context "with valid params" do
+      it "creates a new User" do
+        expect {
+          post :create, {:user => valid_attributes}, valid_session
+        }.to change(User, :count).by(1)
+      end
+
+      it "assigns a newly created user as @user" do
+        post :create, {:user => valid_attributes}, valid_session
+        expect(assigns(:user)).to be_a(User)
+        expect(assigns(:user)).to be_persisted
+      end
+
+      it "redirects to the users list" do
+        post :create, {:user => valid_attributes}, valid_session
+        expect(response).to redirect_to admin_users_path
+      end
+    end
+
+    context "with invalid params" do
+      it "assigns a newly created but unsaved user as @user" do
+        post :create, {:user => invalid_attributes}, valid_session
+        expect(assigns(:user)).to be_a_new(User)
+      end
+
+      it "re-renders the 'new' template" do
+        post :create, {:user => invalid_attributes}, valid_session
+        expect(response).to render_template("new")
+      end
+    end
+  end
+
+  describe "PUT #update" do
+    context "with valid params" do
+      let(:new_attributes) {
+        {
+          first_name: "Steve",
+          last_name: "Rogers",
+          email: "steve@example.com",
+        }
+      }
+
+      it "updates the requested user" do
+        user = User.create! valid_attributes
+        put :update, {:id => user.to_param, :user => new_attributes}, valid_session
+        user.reload
+        expect(user.first_name).to eq("Steve")
+        expect(user.last_name).to eq("Rogers")
+        expect(user.email).to eq("steve@example.com")
+      end
+
+      it "assigns the requested user as @user" do
+        user = User.create! valid_attributes
+        put :update, {:id => user.to_param, :user => valid_attributes}, valid_session
+        expect(assigns(:user)).to eq(user)
+      end
+
+      it "redirects to the users list" do
+        user = User.create! valid_attributes
+        put :update, {:id => user.to_param, :user => valid_attributes}, valid_session
+        expect(response).to redirect_to(admin_users_path)
+      end
+    end
+
+    context "with invalid params" do
+      it "assigns the user as @user" do
+        user = User.create! valid_attributes
+        put :update, {:id => user.to_param, :user => invalid_attributes}, valid_session
+        expect(assigns(:user)).to eq(user)
+      end
+
+      it "re-renders the 'edit' template" do
+        user = User.create! valid_attributes
+        put :update, {:id => user.to_param, :user => invalid_attributes}, valid_session
+        expect(response).to render_template("edit")
+      end
+    end
+  end
+
+  describe "DELETE #destroy" do
+    it "destroys the requested user" do
+      user = User.create! valid_attributes
+      expect {
+        delete :destroy, {:id => user.to_param}, valid_session
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Space inside { missing.<br>Space inside } missing.